app/Model/BuildEmail.php

```diff
@@ -0,0 +1,289 @@
+<?php
+/**
+ * =========================================================================
+ *   Program:   CDash - Cross-Platform Dashboard System
+ *   Module:    $Id$
+ *   Language:  PHP
+ *   Date:      $Date$
+ *   Version:   $Revision$
+ *   Copyright (c) Kitware, Inc. All rights reserved.
+ *   See LICENSE or http://www.cdash.org/licensing/ for details.
+ *   This software is distributed WITHOUT ANY WARRANTY; without even
+ *   the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+ *   PURPOSE. See the above copyright notices for more information.
+ * =========================================================================
+ */
+
+namespace CDash\Model;
+
+use CDash\Collection\BuildEmailCollection;
+use CDash\Config;
+use CDash\Database;
+use CDash\Log;
+use CDash\Messaging\Notification\Email\EmailMessage;
+use CDash\Messaging\Notification\NotificationInterface;
+
+class BuildEmail
+{
+    protected $BuildId;
+    protected $Category;
+    protected $Email;
+    protected $Sent;
+    protected $Time;
+    protected $UserId;
+    private $RequiredFields = ['BuildId', 'UserId', 'Category'];
+
+    /**
+     * @param EmailMessage $notification
+     */
+    public static function SaveNotification(EmailMessage $message)
+    {
+        $collection = $message->getBuildEmailCollection();
+        /** @var BuildEmail $email */
+        foreach ($collection as $emails) {
+            foreach ($emails as $email) {
+                $email->Save();
+            }
+        }
+    }
+
+    public static function Log(NotificationInterface $notification, $sent)
+    {
+        $config = Config::getInstance();
+        $log = Log::getInstance();
+        if ($config->get('CDASH_TESTING_MODE')) {
+            $log->add_log($notification->getRecipient(), 'TESTING: EMAIL', LOG_DEBUG);
+            $log->add_log($notification->getSubject(), 'TESTING: EMAILTITLE', LOG_DEBUG);
+            $log->add_log($notification->getBody(), 'TESTING: EMAILBODY', LOG_DEBUG);
+        } else {
+            $status = $sent ? 'SENT' : 'NOT SENT';
+            $class_name = get_class($notification);
+            $slash_pos = strrpos($class_name, '\\');
+            $pos = $slash_pos ? $slash_pos + 1 : 0;
+            $notification_type = substr($class_name, $pos);
+            $title = $notification->getSubject();
+            $recipient = $notification->getRecipient();
+            $message = "[{$status}] {$notification_type} titled, '{$title}' to {$recipient}";
+            $log->add_log($message, 'BuildEmail::SaveNotification');
+        }
+    }
+
+    /**
+     * @param $userId
+     * @param $buildId
+     * @param $category
+     * @return BuildEmail
+     */
+    public static function GetBuildEmailForUser($userId, $buildId, $category)
```

Comment:
The purpose of this method isn't immediately clear to me either. It's named `Get...` but ends up setting sent = true (among other stuff). Perhaps an example of when this gets called would help.